rust/connlib/tunnel/src/client.rs

```diff
@@ -167,12 +167,11 @@ where
 
     /// Updates the system's dns
     pub fn set_dns(&mut self, new_dns: Vec<IpAddr>) -> connlib_shared::Result<()> {
-        let dns_changed = self.role_state.update_system_resolvers(new_dns);
-
+        self.role_state.update_system_resolvers(new_dns);
```

Comment:
I don't know this code well so I'll just write what I think is happening, like I did for the macOS network PR

So this function has changed to always set the DNS, the logic is moving up from RoleState to whatever this struct is. (ClientTunnel?)

Reply:
well, the function called just below this, which is still in `role_state` checks for changes.